### PR TITLE
Symfony custom extractor

### PR DESCRIPTION
**Pull request by Suvres**
Create custom extractor in symfony and add necessary configuration

### CODE REVIEW

File: symfony/extracting-translations.rst [hunk 1]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
       configs:
         app:
           external_translations_dir: ["%kernel.root_dir%/vendor/friendsofsymfony/user-bundle/Resources/translations"]
+
+
+Create custom extractor
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: Probably better:
> Creating custom extractor

File: symfony/extracting-translations.rst [hunk 2]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+-----------------------
+
+Example method whose argument we want to translate
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: > Example of method...

File: symfony/extracting-translations.rst [hunk 3]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+-----------------------
+
+Example method whose argument we want to translate 
+
+.. code-block:: php
+
+    $this->logger->addMessage("text");
+
+
+Example extractor class that we use to create translations
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: > Example of extractor class...

File: symfony/extracting-translations.rst [hunk 4]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+
+    namespace App\Extractor;
+
+    use PhpParser\Node;
+    use PhpParser\NodeVisitor;
+    use Translation\Extractor\Visitor\Php\BasePHPVisitor;
+    
+    final class MyCustomExtractor extends BasePHPVisitor implements NodeVisitor 
+    {
+
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: This extra line should be removed

File: symfony/extracting-translations.rst [hunk 5]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+    use PhpParser\Node;
+    use PhpParser\NodeVisitor;
+    use Translation\Extractor\Visitor\Php\BasePHPVisitor;
+    
+    final class MyCustomExtractor extends BasePHPVisitor implements NodeVisitor 
+    {
+           
+        /**
+         * {@inheritdoc}
+        */
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: Add one extra space here to align the doc block correctly

File: symfony/extracting-translations.rst [hunk 6]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+        public function beforeTraverse(array $nodes): ?Node
+        {
+            return null;
+        }
+
+        /**
+        * {@inheritdoc}
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: Missing one extra space here and on the next line. And the same in further methods

File: symfony/extracting-translations.rst [hunk 7]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+        */
+        public function afterTraverse(array $nodes): ?Node
+        {
+            return null;
+        }        
+
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: This extra line not needed

File: symfony/extracting-translations.rst [hunk 8]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+            //This "if" check that we have method which interests us
+            if ($name !== "addMessage") {
+                return null;
+            }
+
+            $caller = $node->var;
+            $callerName = isset($caller->name) ? (string) $caller->name : '';
+           
+            //This "if" check that we have xxx->logger->addMessage() 
+            if ($name === 'addMessage' && $callerName === 'logger' && $caller instanceof Node\Expr\MethodCall) {
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: probably `$name === 'addMessage'` check is redundant here? As we already make sure about this fact in the previous if, i.e. in `if ($name !== "addMessage") { return null; }`

File: symfony/extracting-translations.rst [hunk 9]
```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
...
+        */
+        public function beforeTraverse(array $nodes): ?Node
+        {
+            return null;
+        }
+
+        /**
+        * {@inheritdoc}
+        */
+        public function enterNode(Node $node): ?Node
```
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst:75]: Is it OK that this method expects `Node` instance or NULL but we always return only nulls? Don't we want to return `Node` instance at some point?

File: symfony/extracting-translations.rst (hunk 8 above)
Comment (bocharsky-bw) [inline symfony/extracting-translations.rst]: Btw, I'd like to revert this `if` logic and just fast return null that will allow us to move this custom logic out of if below

**Review by bocharsky-bw — COMMENTED** [review]
Hey @Suvres ,

Thank you a lot for writing this missing part in docs! I left some minor comments, mostly about coding style and grammar. I'd be happy to merge this after those tweaks!

File: symfony/extracting-translations.rst (hunk 9 above)
Comment (Suvres) [reply symfony/extracting-translations.rst:75]: Hi, this code is in large part from Your file
https://github.com/php-translation/extractor/blob/master/src/Visitor/Php/Symfony/FlashMessage.php

**Review by bocharsky-bw — APPROVED** [review]
Thank you for adding more changes here!

**Pull request merged**